### DISCUSSION TITLE
Need help wiring PGM to a zone to trigger alarm from homeassistant

### DISCUSSION THREAD

**Discussion by cdaguerre** [ParadoxAlarmInterface/pai#373] [Q&A]
## Alarm system
Magellan MG5050 over IP150

## Environment

HomeAssistant on RPI

### Git branch

dev, 3.2.1

## Question

Hi I installed the PAI interface in HomeAssistant and everything works fine. This integration is impressive!!
I'm trying to trigger the alarm from assistant because I have additional Ubiquiti Sensors and Cameras.

I have 2 PGMs showing up (auto-discovered) and believe they don't do anything.

Apparently the solution would be to have a PGM trigger the alarm, but I have no clue as to what to do ;)
I found this comment from @yozik04, but I have no idea how to wire my zone to the PGM.

              Fully unsupported. Use PGM wired to a zone to trigger alarms.

_Originally posted by @yozik04 in https://github.com/ParadoxAlarmInterface/pai/issues/278#issuecomment-1032291140_
            
Could you point me in the right direction?
I attached a photo of the Magellan...
I just bought this house with this alarm system and I don't know the installer pin-code (which is not 0000).
![IMG_4291](https://github.com/ParadoxAlarmInterface/pai/assets/4642448/f8432fda-bb24-4eed-aadd-0f9933274219)

**Comment by cdaguerre** [comment]
https://github.com/ParadoxAlarmInterface/pai/issues/346 seems related

**Reply by yozik04** [reply]
Wiring a 12V PGM to a Zone through 1kOhm resistor might work.

**Reply by MomosX** [reply]
Hi ! Happy New Year !!!
I am ressurecting an old topic as i've stumbled uppon the same question.  I have an EVO192 connected with PAI to HA.  Trying to make a Panic button in HA.  I havce connected a single wire between PGM3 and zone 8 using an inline 1k resistor.  The zone is set as 24h Burglary. It is not included in any area.
![image](https://github.com/user-attachments/assets/11ea7a6b-c79d-47a9-a527-bc121eb57f36)

![image](https://github.com/user-attachments/assets/b51c3301-f131-4782-b24f-b230014cccd7)

But pressing the PGM3 in HA does nothing.  I see it in the logs that "PGM command: "on" succeeded" but nothing happenes on the other side.  I tried setting the PGM as NO or NC, same result.  If i set it to NC in babyware, there is a voltage of 1.2V between Zone8 and Com.  
when i activate PGM3 in HA the voltage across Zone8 rises to 4.26-4.27.  If i set the PGM3 as NO it's the reverse.
Activating the PGM3 from Babyware has no effect also.  

I'm sure i must be doing something wrong.. i measured the voltage between another zone (PIR) and the com - it's 0 when the PIR is clear and 4.26 when the PIR gets triggered.

Anyone has any idea what am i doing wrong please ? Are there any other settings i need to do in babyware ?
Thank you !

PS: it seems to me it has something to do with area asignement.   If i assign Zone8 to an area and activate PGM3 the alarm starts.  But uppon cancelling the alarm, the Zone8 will not return to Closed status - it will stay opened (green in babywire), so further activations of PGM3 do nothing.  
The only way i can make it return to Closed is if i remove Zone8 from the assigned area.
If i don't assign an area to Zone8, activating PGM3 does nothing.   I hope this makes sense , for me it doesnt :(
I have tried setting the PGM3 as Normal Open and Normal Closed.  Same thing. Zone8 won't reset unless i take it out of the Area. 
I tried to set Zone8 to other Areas, same thing.

**Reply by yozik04** [reply]
Did you wire PGM3 on EVO192 or on a separate extension module? Same question is your Zone 8 on EVO192 or extension module?

**Reply by MomosX** [reply]
PGM3 is on EVO192 as PAI can't see the PGMs from extension module, and Zone8 is wired on an extensoin module with 8 Zones.  I don't think it should make any difference ?

**Reply by yozik04** [reply]
Ok. Then it looks correct. I use all PGMs available on the panel and controlling from PAI works correctly for PGMs that are not binded to any activation events. I personally did not wire any PGMs to zones so I was speculating there. If that does not work you can use a transistor and use PGM as a gate. But firstly I'd figure out why PGM does not return to initial setting after the timer. In my case it always returns fine.

**Reply by MomosX** [reply]
The PGM returns to the initial setting after the timer. always. It's the zone the doesnt - remains open so any future activation of the pgm does nothing.  As soon as i remove the Zone8 from the Area where it is assigned (partition), it returns to the closed state. 

So i was thinking maybe there are some settings in babyware which i did not make.

**Reply by yozik04** [reply]
I don't think there is any setting that controls that. I gave a glimpse into Babyware and have not found anything that seemed to be useful. But I am not an installer, so I am speculating. You still can try a transistor with 1k Ohm EOL resistor.

**Reply by MomosX** [reply]
Thank you very much for the answer.  I have a 1k EOL resitor now on the wire between Zone8 and PGM3.  I'll take any route towards making those PGM's work.   Can you please elaborate on the transistor part? like for non tech savvy folks , which i certainly am.

**Reply by yozik04** [reply]
If it is simpler for you with a relay, do with a relay =)
